Cannot build native-source after adding container-image-buildpack extension.

### Describe the bug

Cannot use native-source packaging when container image extension is added to a project.

### Expected behavior

The maven pack task will pass without error.

### Actual behavior

The maven package task fails:
```
./mvnw package -DskipTests=true -Dmaven.javadoc.skip=true -Dquarkus.package.type=native-sources
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< cz.mvasek:quarkus-buildpack >---------------------
[INFO] Building quarkus-buildpack 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ quarkus-buildpack ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:generate-code (default) @ quarkus-buildpack ---
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ quarkus-buildpack ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:generate-code-tests (default) @ quarkus-buildpack ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ quarkus-buildpack ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /tmp/tmp.DgIJNqy34r/quarkus-buildpack/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ quarkus-buildpack ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ quarkus-buildpack ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ quarkus-buildpack ---
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:build (default) @ quarkus-buildpack ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.333 s
[INFO] Finished at: 2022-03-16T14:32:59+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.4.Final:build (default) on project quarkus-buildpack: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.deployment.ContainerImageProcessor#failForNativeSources threw an exception: java.lang.IllegalArgumentException: The Container Image extensions are incompatible with the 'native-sources' package type.
[ERROR] 	at io.quarkus.container.image.deployment.ContainerImageProcessor.failForNativeSources(ContainerImageProcessor.java:26)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### How to Reproduce?

1. Create simple RESTEasy app using https://code.quarkus.io/?e=resteasy
2. Add the container-image-buildpack extension `./mvnw quarkus:add-extension -Dextensions="container-image-buildpack"` to the project.
3. Try to build project with `native-source`: `./mvnw package -DskipTests=true -Dmaven.javadoc.skip=true -Dquarkus.package.type=native-sources`

The build fails. Maybe I could understand if that happended if `quarkus.container-image.build=true`, but it happens even when the container image build is not in use.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java
+++ b/extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java
@@ -1,6 +1,8 @@
 
 package io.quarkus.container.image.docker.deployment;
 
+import static io.quarkus.container.image.deployment.util.EnablementUtil.buildContainerImageNeeded;
+import static io.quarkus.container.image.deployment.util.EnablementUtil.pushContainerImageNeeded;
 import static io.quarkus.container.util.PathsUtil.findMainSourcesRoot;
 
 import java.io.BufferedReader;
@@ -73,12 +75,9 @@ public class DockerProcessor {
             @SuppressWarnings("unused") // used to ensure that the jar has been built
             JarBuildItem jar) {
 
-        if (containerImageConfig.isBuildExplicitlyDisabled() && !containerImageConfig.isPushExplicitlyEnabled()) {
-            return;
-        }
-
-        if (!containerImageConfig.isBuildExplicitlyEnabled() && !containerImageConfig.isPushExplicitlyEnabled()
-                && !buildRequest.isPresent() && !pushRequest.isPresent()) {
+        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
+        if (!buildContainerImage && !pushContainerImage) {
             return;
         }
 
@@ -104,7 +103,7 @@ public class DockerProcessor {
         ImageIdReader reader = new ImageIdReader();
         String builtContainerImage = createContainerImage(containerImageConfig, dockerConfig, containerImageInfo, out, reader,
                 false,
-                pushRequest.isPresent(), packageConfig);
+                pushContainerImage, packageConfig);
 
         // a pull is not required when using this image locally because the docker strategy always builds the container image
         // locally before pushing it to the registry
@@ -125,12 +124,9 @@ public class DockerProcessor {
             // used to ensure that the native binary has been built
             NativeImageBuildItem nativeImage) {
 
-        if (containerImageConfig.isBuildExplicitlyDisabled() && !containerImageConfig.isPushExplicitlyEnabled()) {
-            return;
-        }
-
-        if (!containerImageConfig.isBuildExplicitlyEnabled() && !containerImageConfig.isPushExplicitlyEnabled()
-                && !buildRequest.isPresent() && !pushRequest.isPresent()) {
+        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
+        if (!buildContainerImage && !pushContainerImage) {
             return;
         }
 
@@ -147,7 +143,7 @@ public class DockerProcessor {
 
         ImageIdReader reader = new ImageIdReader();
         String builtContainerImage = createContainerImage(containerImageConfig, dockerConfig, containerImage, out, reader, true,
-                pushRequest.isPresent(), packageConfig);
+                pushContainerImage, packageConfig);
 
         // a pull is not required when using this image locally because the docker strategy always builds the container image
         // locally before pushing it to the registry

--- a/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java
+++ b/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java
@@ -1,5 +1,7 @@
 package io.quarkus.container.image.jib.deployment;
 
+import static io.quarkus.container.image.deployment.util.EnablementUtil.buildContainerImageNeeded;
+import static io.quarkus.container.image.deployment.util.EnablementUtil.pushContainerImageNeeded;
 import static io.quarkus.container.util.PathsUtil.findMainSourcesRoot;
 
 import java.io.IOException;
@@ -134,10 +136,8 @@ public class JibProcessor {
             Optional<AppCDSResultBuildItem> appCDSResult,
             BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
 
-        Boolean buildContainerImage = containerImageConfig.isBuildExplicitlyEnabled()
-                || (buildRequest.isPresent() && !containerImageConfig.isBuildExplicitlyDisabled());
-        Boolean pushContainerImage = containerImageConfig.isPushExplicitlyEnabled()
-                || (pushRequest.isPresent() && !containerImageConfig.isPushExplicitlyDisabled());
+        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
         if (!buildContainerImage && !pushContainerImage) {
             return;
         }
@@ -167,7 +167,7 @@ public class JibProcessor {
 
         artifactResultProducer.produce(new ArtifactResultBuildItem(null, "jar-container",
                 Map.of("container-image", container.getTargetImage().toString(), "pull-required",
-                        pushContainerImage.toString())));
+                        Boolean.toString(pushContainerImage))));
     }
 
     @BuildStep(onlyIf = { IsNormalNotRemoteDev.class, JibBuild.class, NativeBuild.class })
@@ -181,10 +181,8 @@ public class JibProcessor {
             Optional<UpxCompressedBuildItem> upxCompressed, // used to ensure that we work with the compressed native binary if compression was enabled
             BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
 
-        Boolean buildContainerImage = containerImageConfig.isBuildExplicitlyEnabled()
-                || (buildRequest.isPresent() && !containerImageConfig.isBuildExplicitlyDisabled());
-        Boolean pushContainerImage = containerImageConfig.isPushExplicitlyEnabled()
-                || (pushRequest.isPresent() && !containerImageConfig.isPushExplicitlyDisabled());
+        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
         if (!buildContainerImage && !pushContainerImage) {
             return;
         }
@@ -207,7 +205,7 @@ public class JibProcessor {
 
         artifactResultProducer.produce(new ArtifactResultBuildItem(null, "native-container",
                 Map.of("container-image", container.getTargetImage().toString(), "pull-required",
-                        pushContainerImage.toString())));
+                        "" + pushContainerImage)));
     }
 
     private JibContainer containerize(ContainerImageConfig containerImageConfig,

--- a/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java
+++ b/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java
@@ -1,11 +1,15 @@
 package io.quarkus.container.image.deployment;
 
+import static io.quarkus.container.image.deployment.util.EnablementUtil.*;
+
 import java.util.Collections;
 import java.util.Optional;
 
 import org.jboss.logging.Logger;
 
+import io.quarkus.container.spi.ContainerImageBuildRequestBuildItem;
 import io.quarkus.container.spi.ContainerImageInfoBuildItem;
+import io.quarkus.container.spi.ContainerImagePushRequestBuildItem;
 import io.quarkus.container.spi.FallbackContainerImageRegistryBuildItem;
 import io.quarkus.container.spi.ImageReference;
 import io.quarkus.deployment.Capabilities;
@@ -22,9 +26,14 @@ public class ContainerImageProcessor {
     private static final Logger log = Logger.getLogger(ContainerImageProcessor.class);
 
     @BuildStep(onlyIf = NativeSourcesBuild.class)
-    void failForNativeSources(BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
-        throw new IllegalArgumentException(
-                "The Container Image extensions are incompatible with the 'native-sources' package type.");
+    void failForNativeSources(ContainerImageConfig containerImageConfig,
+            Optional<ContainerImageBuildRequestBuildItem> buildRequest,
+            Optional<ContainerImagePushRequestBuildItem> pushRequest,
+            BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
+        if (buildOrPushContainerImageNeeded(containerImageConfig, buildRequest, pushRequest)) {
+            throw new IllegalArgumentException(
+                    "The Container Image extensions are incompatible with the 'native-sources' package type.");
+        }
     }
 
     @BuildStep

--- a/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/util/EnablementUtil.java
+++ b/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/util/EnablementUtil.java
@@ -1,0 +1,29 @@
+package io.quarkus.container.image.deployment.util;
+
+import java.util.Optional;
+
+import io.quarkus.container.image.deployment.ContainerImageConfig;
+import io.quarkus.container.spi.ContainerImageBuildRequestBuildItem;
+import io.quarkus.container.spi.ContainerImagePushRequestBuildItem;
+
+public class EnablementUtil {
+
+    public static boolean buildContainerImageNeeded(ContainerImageConfig containerImageConfig,
+            Optional<ContainerImageBuildRequestBuildItem> buildRequest) {
+        return containerImageConfig.isBuildExplicitlyEnabled()
+                || (buildRequest.isPresent() && !containerImageConfig.isBuildExplicitlyDisabled());
+    }
+
+    public static boolean pushContainerImageNeeded(ContainerImageConfig containerImageConfig,
+            Optional<ContainerImagePushRequestBuildItem> pushRequest) {
+        return containerImageConfig.isPushExplicitlyEnabled()
+                || (pushRequest.isPresent() && !containerImageConfig.isPushExplicitlyDisabled());
+    }
+
+    public static boolean buildOrPushContainerImageNeeded(ContainerImageConfig containerImageConfig,
+            Optional<ContainerImageBuildRequestBuildItem> buildRequest,
+            Optional<ContainerImagePushRequestBuildItem> pushRequest) {
+        return buildContainerImageNeeded(containerImageConfig, buildRequest)
+                || pushContainerImageNeeded(containerImageConfig, pushRequest);
+    }
+}

